components/storage/blockdevice/COMPONENT_QSPIF/QSPIFBlockDevice.cpp

```diff
@@ -132,23 +181,24 @@ int QSPIFBlockDevice::init()
         tr_debug("QSPIFBlockDevice csel: %d", (int)_csel);
     } else if (_unique_device_status == -1) {
         tr_error("QSPIFBlockDevice with the same csel(%d) already exists", (int)_csel);
-        return QSPIF_BD_ERROR_DEVICE_NOT_UNIQE;
+        return QSPIF_BD_ERROR_DEVICE_NOT_UNIQUE;
     } else {
         tr_error("Too many different QSPIFBlockDevice devices - max allowed: %d", QSPIF_MAX_ACTIVE_FLASH_DEVICES);
         return QSPIF_BD_ERROR_DEVICE_MAX_EXCEED;
     }
 
-    uint8_t vendor_device_ids[4];
-    size_t data_length = 3;
     int status = QSPIF_BD_ERROR_OK;
     uint32_t basic_table_addr = 0;
     size_t basic_table_size = 0;
     uint32_t sector_map_table_addr = 0;
     size_t sector_map_table_size = 0;
-    int qspi_status = QSPI_STATUS_OK;
 
     _mutex.lock();
 
+    // All commands other than Read and RSFDP use default 1-1-1 bus mode (Program/Erase are constrained by flash memory performance more than bus performance)
+    _qspi.configure_format(QSPI_CFG_BUS_SINGLE, QSPI_CFG_BUS_SINGLE, _address_size, QSPI_CFG_BUS_SINGLE,
```

Comment:
Do you need to check return value from `configure_format()`?

Reply:
Yes, the return value needs to be checked for configure_format across the board, but that is handled in #11603 (applying the check to additional/relocated configure_format calls would be handled during the rebase of whichever PR goes in second).